lib/Auth/AuthHandler.php

```diff
@@ -0,0 +1,30 @@
+<?php
+namespace Spark\Auth;
+
+use Auryn\Injector;
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
+
+class AuthHandler
+{
+
+    protected $authenticator;
+    protected $injector;
+
+    public function __construct(AbstractAuthenticator $authenticator)
+    {
+        $this->authenticator = $authenticator;
+    }
+
+    public function __invoke(Request $request, Response $response, callable $next)
+    {
+
+        $token = $this->authenticator->getToken($request);
+
+        if ($token && !$this->authenticator->isValid($token)) {
+            throw new \Exception($this->authenticator->getErrorMessage());
```

Comment:
I would actually let the authenticator throw the exception:

```php
if (/* failed */) {
    $this->authenticator->failure($token);
}
```

One pattern I've used in the past that makes this easier:

```php
public function isValid(...) { ... }
public function ensureValid(...)
{
    if (!$this->isValid()) {
        throw new FooException(...);
    }
}
```

Which allows you to write one-liners:

```php
$this->authenticator->ensureValid($token); // will throw an exception if not valid #winning
```